apps/advisor/pkg/app/utils.go

```diff
@@ -83,18 +84,15 @@ func processCheck(ctx context.Context, client resource.Client, obj resource.Obje
 	}
 	// Run the steps
 	steps := check.Steps()
-	errs := []advisorv0alpha1.CheckReportError{}
-	for _, step := range steps {
-		stepErrs, err := step.Run(ctx, &c.Spec, items)
-		if err != nil {
-			setErr := setStatusAnnotation(ctx, client, obj, "error")
-			if setErr != nil {
-				return setErr
-			}
-			return fmt.Errorf("error running step %s: %w", step.Title(), err)
+	errs, err := runStepsInParallel(ctx, &c.Spec, steps, items)
```

Comment:
Just a thought after seeing the result of these calls returning two "errors".  Maybe we should rename `advisorv0alpha1.CheckReportError` to `advisorv0alpha1.CheckReport` to avoid ambiguity?

Reply:
Those names are autogenerated from the CUE files so not really something we could change. Also, `advisorv0alpha1.CheckReport` already means the struct that contains "errors" and "count" so not something that we could use.

What we could do is change the word "error" for something like "failure". I can do that in a different PR to avoid more changes here. WDYT?

Reply:
Yeah I think that's reasonable. Anything to avoid conflating it with being an actual "error" that needs to be handled 👍